src/pages/quiz/QuizVariants.tsx

```diff
@@ -117,18 +109,7 @@ const ActiveQuiz: React.FC = () => {
   const editFinishedQuiz = useCallback(() => {
     return ResourceService.editFinishedQuiz(finishedQuizId, {
       grade,
-      results: items.map(({ text, answers, _id }) => {
-        return {
-          question: text,
-          answers: answers.map(({ text, isCorrect }) => {
-            return {
-              text,
-              isCorrect,
-              isChosen: data[_id]?.includes(text) ?? false
-            }
-          })
-        }
-      })
+      results: formatQuizResults(data, items)
```

Comment:
Can you use `useMemo` result here?
```suggestion
      results: mappedResults
```